examples/vision/learnable_resizer.py

```diff
@@ -0,0 +1,273 @@
+Author: [Sayak Paul](https://twitter.com/RisingSayak)
+Date created: 2021/04/30
+Last modified: 2021/04/30
+Description: How to optimally learn representations of images for a given resolution.
+"""
+"""
+It is a common belief that if we constrain vision models to perceive things as humans do,
+their performance can be improved. For example, in [this
+work](https://arxiv.org/abs/1811.12231), Geirhos et. al showed that the vision models
+pre-trained on the ImageNet-1k dataset are biased toward texture whereas human beings
+mostly use the shape descriptor to develop a common perception. But does this belief
+always apply especially when it comes to improving the performance of vision models? 
+
+It turns out it may not always be the case. When training vision models, it is common to
+resize images to a lower dimension ((224x224), (299x299), etc.) to allow mini-batch
+learning and also to keep up the compute limitations.  We generally make use of image
+resizing methods like **bilinear interpolation** for this step and the resized images do
+not lose much of their perceptual character to the human eyes. In [Learning to Resize
+Images for Computer Vision Tasks](https://arxiv.org/abs/2103.09950v1), Talebi et. al show
+that if we try to optimize the perceptual quality of the images for the vision models
+rather than the human eyes, their performance can further be improved. They investigate
+the following question: 
+
+**For a given image resolution and a model, how to best resize the given images?**
+
+As shown in the paper, this idea helps to consistently improve the performance of the
+common vision models (pre-trained on ImageNet-1k) like DenseNet121, ResNet50,
+MobileNetV2, and EfficientNets. In this example, we will implement the learnable image
+resizing module as proposed in the paper and demonstrate that on the
+[Cast-Vs-Dogs dataset](https://www.microsoft.com/en-us/download/details.aspx?id=54765)
+using the [DenseNet121](https://arxiv.org/abs/1608.06993) model. 
+
+This example requires TensorFlow 2.4 or higher.
+"""
+
+"""
+## Setup
+"""
+
+from tensorflow.keras import layers
+from tensorflow import keras
+import tensorflow as tf
+
+import tensorflow_datasets as tfds
+
+tfds.disable_progress_bar()
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+"""
+## Define hyperparameters
+"""
+
+"""
+In order to facilitate mini-batch learning, we need to have a fixed shape for the images
+inside a given batch. This is why an initial resizing is required. We first resize all
+the images to (300x300) shape and then learn their optimal representation for the
+(224x224) resolution. 
+"""
+
+INP_DIM = (300, 300)
+TARGET_DIM = (224, 224)
+INTERPOLATION = "bilinear"
+
+AUTO = tf.data.AUTOTUNE
+BATCH_SIZE = 64
+EPOCHS = 5
+ALPHA = 0.2
+
+"""
+For this example, we will use the bilinear interpolation but the learnable image resizer
+module is not dependent on any specific interpolation method. We can use other ones (such
+as bicubic) as well. 
```

Comment:
Suggestion: refactor, 80 char limit 

```suggestion
In this example, we will use the bilinear interpolation but the learnable image
resizer module is not dependent on any specific interpolation method. We can
also use others, such as bicubic.
```